### PR TITLE
Update parse_it.py to 18.8.12

### PR DESCRIPTION
**Pull request by aleale99**
## Description
with commit 1984 completes the work on the it_it version of format.py and parse.py to align the functions to  English version 18.8.12
but now I think everything has already been overcome after the excellent work of  ChristopherRogers1991

## How to test
added new tests and updated old ones in test_parse_it.py

## Contributor license agreement signed?
CLA [X] (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

File: mycroft/util/lang/parse_it.py [hunk 1]
```diff
@@ -204,217 +215,373 @@ def isFractional_it(input_str, short_scale=False):
...
+            components[0] = components[0][1:]
+        word = str(extractnumber_long_it(components[0])) + \
+            '*1000' + str(components[1])
+
+    # gestione eccezioni
+    if word.startswith('*'):
+        word = word[1:]
+
+    try:
+        value = eval(word)
```
[the remark this answers is not in the document]
Comment (forslund) [reply mycroft/util/lang/parse_it.py]: Why is eval used here?

**Comment by forslund** [comment]
Hi, can you check the codacy issues?

I'd also like to have comments and docstrings in english to make it easier to review.

File: mycroft/util/lang/parse_it.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aleale99) [reply mycroft/util/lang/parse_it.py]: in Italian long numbers can be written in a single word  something like   "thirteenmillioneighthundredfortysixthousandsevenhundredeightytwo"  
I replace the elements of the string with the algebraic equivalents, so eval can calculate it.
eval fails correctly in the case of word that is not a number
for example 'qualcuno' become 'qualc+1' (something like 'someone' become 'some+1') o 'tremore' --> '+3more' (like 'tension' --> '+10sion')

I thought it was an easy way to deal with numbers of any length
Is it a inappropriate use of eval?

[the remark this answers is not in the document]
Comment (forslund) [reply mycroft/util/lang/parse_it.py]: Well eval executes arbitrary code, so it's not really a good function to use. Can't you use something like
```python
# Check if all tokens are numbers
if all([s.isdecimal() for s in word.split("+")]):
    value = sum([int(s) for s in word.split("+")])
else:
    value = None
```

**Comment by forslund** [comment]
Great job with this @aleale99, thank you! It looks good as far as I can see. Merging

**Pull request merged**